pkg/arvo/app/link-store.hoon

```diff
@@ -0,0 +1,172 @@
+::  link: social bookmarking
+::
+::    the paths under which links are submitted are generally expected to
+::    correspond to existing group paths. for strictly-local collections of
+::    links, arbitrary paths are probably fair game, but could trip up
+::    primitive ui implementations.
+::
+::  scry and subscription paths:
+::
+::    /local-pages/[some-group]    all pages we saved by recency
+::    /submissions/[some-group]    all submissions by recency
+::
+/+  *link, default-agent, verb
+::
+|%
++$  state-0
+  $:  %0
+      by-group=(map path links)
+      by-site=(map site (list [path submission]))
+  ==
+::
++$  links
+  $:  ::NOTE  all lists by recency
+      =submissions
+      ours=pages
+  ==
+::
++$  card  card:agent:gall
+--
+::
+=|  state-0
+=*  state  -
+::
+%+  verb  &
+^-  agent:gall
+=<
+  |_  =bowl:gall
+  +*  this  .
+      do    ~(. +> bowl)
```

Comment:
do? Perhaps this could be better named.

Reply:
As in the verb. My general structuring of new-gall apps is that the agent core is just handling gall interface tasks, and then the core below it actually _does_ all the application logic. Hence `do`. I think it reads quite nicely in most cases, `get-local-pages:do` etc. Happy to listen to suggestions for better names, if you have them.

Reply:
Another option is liberal use of |^.  For example, +do-action, +add-page, and +hear-submission could be in a |^ of on-poke.  The other two are used in both +on-peek and +on-watch, though.  You have to weigh the advantages of tighter scoping and easier state management against the reduced flexibility if you later want to call the function from another arm.